go-controller/pkg/ovn/ovn.go

```diff
@@ -377,112 +375,6 @@ func (oc *Controller) Run(wg *sync.WaitGroup) error {
 	return nil
 }
 
-type eventRecord struct {
-	Data     [][]interface{} `json:"Data"`
-	Headings []string        `json:"Headings"`
-}
-
-type emptyLBBackendEvent struct {
-	vip      string
-	protocol kapi.Protocol
-	uuid     string
-}
-
-func extractEmptyLBBackendsEvents(out []byte) ([]emptyLBBackendEvent, error) {
-	events := make([]emptyLBBackendEvent, 0, 4)
```

Comment:
This isn't "allocating a fixed number of events struct", it's just pre-allocating space so that we can make up to 4 `append` calls without needing to reallocate the slice. It's an optimization. It's good.

Reply:
hmm then I think that what we want is `events := make([]emptyLBBackendEvent, 4)`  that is the same as `events := make([]emptyLBBackendEvent, 4,4)`

> A slice can be created with the built-in function called make, which has the signature,
> func make([]T, len, cap) []T

https://blog.golang.org/slices-intro

Reply:
anyway, I think that this is premature optimization, I have doubts this really suppose an impact in performance

Reply:
`make([]X, 0, 4)` creates an array that has 0 elements but has enough space to store 4 elements. If you append to it 4 times, you'll get an array that has 4 elements, and the runtime won't have had to call malloc() again after the initial allocation.

`make([]X, 0)` creates an array that has 0 elements and no space to store any elements. If you append to it 4 times, it will have to call malloc() again at least once, and possibly as many as 4 times.

`make([]X, 4)` or `make([]X, 4, 4)` creates an array that has 4 `nil` elements. If you append to it 4 times, you'll have an array with 4 `nil` elements followed by 4 non-`nil` elements, which is totally not what you wanted.

Reply:
ahh, I was confused with the `len` and `cap` meaning, will keep as is then
Thanks